### DISCUSSION TITLE
`matcher.send`似乎不拒绝空字符串

### DISCUSSION THREAD

**Discussion by fu050409** [nonebot/discussions#411] [想法]
我使用Onebot v11作为适配器。

我的项目中有一部分功能是需要展示目前的某些数据，通过for循环来拼接字符串。

然而当数据为空的时候，for循环会被略过，变量`reply`存储了一个长度为`0`的字符串，`matcher.send(reply)`并没有抛出异常，而是硬生生`send`了一个空字符串。

我在电脑端的QQ上查看机器人的消息，发现它提示`该消息暂不支持在电脑端查看，请在手机端查看`。emm，我不是直接复制粘贴的但大概是这个意思。然后我打开手机，它又告诉我`该消息暂不支持查看`。我当时一脸懵。调试了几次之后才发现我发送了一个空消息。

由于我没做过其它平台的机器人，不太清楚其它协议是否有空消息发送的情景，但是在QQ这边的确会出现问题，而且会提高封号的风险，不知道`Nonebot2`这边需不需要加一个空字符串检定？

**Comment by yanyongyu** [comment]
空字符串不代表是不能发的，nb并不是为QQ服务的

**Reply by fu050409** [reply]
多谢大佬，我了解了。

不过我建议可以把这个写进文档作为一个特别提示，提示一下使用`Onebot v11`做QQ机器人的人避雷。

**Reply by yanyongyu** [reply]
onebot也是个多平台协议。。

**Reply by fu050409** [reply]
> onebot也是个多平台协议。。

这样啊……我是从nb这边才接触Onebot的，是我着相了。

我的意思是觉得腾讯的QQ机器人这边问题确实比较多，~~觉得可以在文档里面提醒一下~~。不过大佬觉得不用那就算了吧
